### ISSUE TITLE
MaskArray.take does not work with scalar indices

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#7585]
Python 3.5, numpy 1.10.1

```python
>>> m = np.ma.array([1, 2, 3, 4], mask=[True, 0, 0, 0])

>>> m[0]
masked
>>> m.take(0)  #This is wrong
1

>>> m[[0,1]]
masked_array(data = [-- 2],
             mask = [ True False],
       fill_value = 999999)
>>> m.take([0, 1])
masked_array(data = [-- 2],
             mask = [ True False],
       fill_value = 999999)
```

**Comment by charris** [comment]
Interesting. @ahaldane Thoughts?

**Comment by eric-wieser** [comment]
Fixed by adding a `np.asarray` and a `[...]`, I think

**Comment by eric-wieser** [comment]
The underlying issue is that `np.int32(1).view(MaskedArray)` does not return a `MaskedArray`!

**Comment by eric-wieser** [comment]
Workaround: `m.take(keys[np.newaxis])[0]` in place of `m.take(keys)`

**Comment by charris** [comment]
Closed by #7586.
